src/common/file_retriever.py

```diff
@@ -35,7 +37,7 @@ def split_protocol(url):
         return None, None
 
 
-def _s3_save(protocol, source_loc, dest_obj, auth="saml"):
+def _s3_save(protocol, source_loc, dest_obj, auth, logger):
```

Comment:
Couldn't auth still be a kwarg? It looks like you expect it to be None in some cases anyway?

Reply:
You can always pass it as a kwarg anyway even if it isn't defined as one. Anyway the internal calls are all underscores, and the get call has it as a kwarg. I don't know. Yes? 😆 But for what?

Reply:
Well I guess if these individual save methods are never going to be used independently of the file retriever then it doesn't matter. If they are, then auth should be optional. I'm ok with it the way it is